staging/src/k8s.io/apiserver/pkg/authentication/request/x509/x509.go

```diff
@@ -71,6 +85,8 @@ func (a *Authenticator) AuthenticateRequest(req *http.Request) (user.Info, bool,
 		}
 	}
 
+	remaining := req.TLS.PeerCertificates[0].NotAfter.Sub(time.Now())
+	clientCertificateExpirationGauge.WithLabelValues(req.TLS.PeerCertificates[0].Subject.CommonName).Set(float64(remaining / time.Second))
```

Comment:
the cardinality on this is way too high. a label per client and essentially infinite values. see the warning at the bottom of https://prometheus.io/docs/practices/naming/